### DISCUSSION TITLE
Antismash output file

### DISCUSSION THREAD

**Discussion by pdhrati02** [antismash/antismash#433] [General]
Hi, I have about 1000 gbk files from running antismash on assembled genome.
I tried parsing these results to get a tsv file containing count information for each BCG for each assembled genome in the following manner:

for files in glob.glob("*.gbk"):
    out_files = "products/"+files.replace(".gbk","_output.tsv")
    cluster_out = open(out_files, "w")

# Extract Cluster info, write to file
    for seq_record in SeqIO.parse(files, "genbank"):
     for seq_feat in seq_record.features:
      if seq_feat.type == "protocluster":
       cluster_number = seq_feat.qualifiers["protocluster_number"][0].replace(" ","_").replace(":","")
       cluster_type = seq_feat.qualifiers["product"][0]

       cluster_out.write("#"+cluster_number+"\tCluster Type:"+cluster_type+"\n") 

However, while doing so while most of the files work fine, some gbk files give an empty output file. Can someone help me with why is this happening?
I am losing about 360 files in this manner out of 1000! So before going ahead I am trying to understand the reason behind this. Is there any other way to find if the gbk file is right? Why do these files dont show any BCG?
For your reference please find attached one such file which gives an empty output with the above code (copied the contents to a txt file as could not upload a gbk file).

Any help will be appreciated. 
Thank you in advance
Best
DP
[gbk_example.txt](https://github.com/antismash/antismash/files/8366135/gbk_example.txt)

**Comment by SJShaw** [comment]
None of the records in your example contain any protoclusters, which is why you aren't able to get protoclusters from them. As a genome mining tool, antiSMASH really needs complete genomes to get decent results. This particular input is much, much too fragmented to get decent results and it's not surprising no protoclusters are detected.

If the other 359 files that have no protoclusters are similarly fragmented, then it's likely the same problem as your example file. To check, you can search for the word "protocluster" in those files, if it isn't present, you won't be able to extract any protoclusters from them.

**Reply by pdhrati02** [reply]
Hi @SJShaw Thank you very much for your response, your explanation is very helpful, but I want to ask you if there are any more factors that would affect this process of protocluster generation? 
Because I checked the number of contigs for all my assembled genomes (in the mags/bins) and I found a few examples like the file attached, where there a huge number of contigs present (meaning many fragments as I understand now) and still it gives protoclusters. 

Infact the quality of the genome (stats based on completeness and contamination) I shared earlier was better than the one I am sharing now, so I just want to ask you what could be the other factors which would affect this process of protocluster formation. 

Thank you once again for your explanation
Best
DP
[gbk_example2.txt](https://github.com/antismash/antismash/files/8370097/gbk_example2.txt)

**Reply by SJShaw** [reply]
Luck is the major factor involved with this level of fragmentation.

It's virtually impossible to find most cluster types in contigs this small, and those small types you might find (like the RiPPs here) are almost certainly missing regulatory genes and transporters and all the other parts that real clusters typically have.

This kind of input just isn't suitable for antiSMASH.

**Reply by pdhrati02** [reply]
Hi @SJShaw Thank you very much, but when you say this kind of input is unsuitable for antiSMASH do you means MAGs or small contigs are not suitable?
Also a separate question altogether: Would it always be better to normalize the antiSMASH results before use? Any rule of thumb one must follow?

Thank you once again for your explanation.
Best
DP

**Reply by SJShaw** [reply]
The larger the contig the better the results. The smaller the contig the worse the results (unless you've specifically taken a small section that you know contains your area of interest). MAGs are generally all extremely small contigs.

I don't know what you mean by normalising antiSMASH results, sorry.

**Reply by pdhrati02** [reply]
Hi @SJShaw Thank you very much for your explanations, they are very helpful.
Best
DP
